Feature/resource cards

kylemhudson opened the pull request:
**Ready for the new Learning Resources page to be reviewed. Here's what this PR accomplishes:** 

New layout for the Learning Resources page (screenshot attached).

- Create a Resource Card component (ResourceCard.js) with an image, title, description, and tag. Image and title are clickable to the external site. 
- Reformat Learning Resources (LearningResources.js) page to accommodate (responsively) the new cards. 
- Create /assets/images/resources folder with all Resource Card images. All images are scaled and optimized below 100kb.

**Future improvements:** 

- I'd like to sort the page by tag (All, Free, Paid). Tags at the moment are static/decorative

![resources_page_update](https://user-images.githubusercontent.com/14058890/88469171-539a0e00-cebc-11ea-9bd7-f988f13d7723.jpg)

forresthayes commented on src/pages/LearningResources.js:
```diff
@@ -1,231 +1,178 @@
+    <div className='container m-auto max-w-screen-xl align-middle mt-32 sm:max-w-screen-lg'>
+      <div class='relative px-12 text-center mx-auto my-10 sm:px-40 lg:px-32'>
+        <h2 class='text-2xl leading tracking-tight font-extrabold text-blue-900 sm:leading-10 sm:text-3xl'>
+          Lerning Resources
```
Quick typo fix - 'Learning'

kylemhudson replied:
Thanks, I'll give it a fix￼

kylemhudson commented:
@drewclem @forresthayes [Commit: 6bf0147](https://github.com/drewclem/protege/pull/107/commits/6bf0147fbdfb8ac23efa9435a6172886ddc4ad7c) addresses: 

- Fixed Learning typo on the LearningResources.js page 
- Updated Traversy Media thumbnail to not include the YouTube side navigation in the shot

pickleat commented:
Hey @kylemhudson I'm reviewing now!

pickleat commented on src/pages/LearningResources.js:
```diff
@@ -1,231 +1,178 @@
+        <p className='text-base mb-2 text-blue-700 sm:mt-4 sm:px-12 sm:text-base md:text-base'> We've pulled some of the best free, and paid, resources together to help support your journey.
+        </p>
+        <p className='text-base text-blue-700 sm:mt-4 sm:text-base md:text-base'> Got a resource you think we should add to the list?{' '}
+          <a className='underline' href={'https://twitter.com/DevProtege'}>
```
Let's make this a `<Link to='get-in-touch'>`

kylemhudson replied:
This one has been addressed. 

Also added back in the Framer Motion div to match the other pages.

pickleat commented on src/pages/LearningResources.js:
```diff
@@ -1,231 +1,178 @@
+
+        <ResourceCard
+          title='Wes Bos Youtube'
+          resourceimg='wesbos.jpg'
```
This should be `resourceimg='wesbos-youtube.jpg'

kylemhudson replied:
Swapped Wes' image links

pickleat commented on src/pages/LearningResources.js:
```diff
@@ -1,231 +1,178 @@
+          />
+          <ResourceCard
+            title='Wes Bos Courses'
+            resourceimg='wesbos-youtube.jpg'
```
should be `resourceimg='wesbos.jpg'`

kylemhudson replied:
Swapped Wes' image links

pickleat commented on src/components/ResourceCard.js:
```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+
+const ResourceCard = (props) => {
+  const { title, resourceimg, description, resourceurl, tag } = props
```
I think we need to make a change to the props names, for consistency across the code base. The props should change to `resourceImage` and `resourceUrl`. Obviously this will change in the `LearningResources.js` page as well. I hate to be a stickler, but we will need to refactor this.

kylemhudson replied:
Refactored ResourceCard.js props to ```resourceImage``` and ```resourceUrl```

Updated LearningResources.js props to reflect the new component props

pickleat commented on src/pages/LearningResources.js:
```diff
@@ -1,231 +1,178 @@
       </div>
-    </motion.div>
+
+      <div className='content-center flex grid-flow-col flex-wrap'>
```
add class `justify-center` to be: `<div className='content-center flex grid-flow-col flex-wrap justify-center'>` so on mobile the cards will be centered.

kylemhudson replied:
Adjusted Free Resources content div to be ```justify-center```

![iPhone 11_414x896_localhost_ Jul_26_03-49](https://user-images.githubusercontent.com/14058890/88487939-a8d72d80-cf57-11ea-81b0-67d9a2a4cff0.png)

pickleat commented on src/pages/LearningResources.js:
```diff
@@ -1,231 +1,178 @@
+          <h3 className='text-xl text-blue-900 sm:text-2xl md:text-3xl'>Paid Resources</h3>
+        </div>
+
+        <div className='content-center flex grid-flow-col flex-wrap'>
```
Same as the free resources change above. Add `justify-center` to the list of classes.

kylemhudson replied:
Adjusted Paid Resources content div to be ```justify-center```

![paid](https://user-images.githubusercontent.com/14058890/88487973-010e2f80-cf58-11ea-8596-79669fc2250b.png)

kylemhudson commented:
Sweet, thanks. Will get on those updates. Thx for the review.

kylemhudson commented:
@pickleat All changes requested have been resolved. Ready for review. 

Thanks!

drewclem commented:
@kylemhudson This looks solid. I'm going to add a few minor things just for my personal preference - but I'll get this merged in today

kylemhudson commented:
Thanks!
